### ISSUE TITLE
Slurm plugin missing header

### ISSUE THREAD

**Issue by gdolle** [apptainer/singularity#701]
When trying to compile the slurm plugin `--with-slurm`, I get an error with a missing header:

```
singularity.c:32:29: fatal error: lib/singularity.h: No such file or directory  #include "lib/singularity.h"
```

Maybe this include should be removed ? (like here https://github.com/singularityware/singularity/commit/e699be95cdb7c2352f22dfa844c2cec301e0adfe)

## Version of Singularity:
Release 2.3.0

**Comment by kav2k** [comment]
To clarify: this only happens when trying to compile Slurm integration plugin.

**Comment by moskalenko** [comment]
Same here, full error:

Making all in slurm
make[3]: Entering directory `singularity-2.3/src/slurm'
depbase=`echo singularity.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ../../libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../src     -I/opt/slurm/include -MT singularity.lo -MD -MP -MF $depbase.Tpo -c -o singularity.lo singularity.c &&\
        mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../src -I/opt/slurm/include -MT singularity.lo -MD -MP -MF .deps/singularity.Tpo -c singularity.c  -fPIC -DPIC -o .libs/singularity.o
singularity.c:32:29: error: lib/singularity.h: No such file or directory
singularity.c: In function ‘setup_container_environment’:
singularity.c:66: warning: incompatible implicit declaration of built-in function ‘snprintf’
singularity.c: In function ‘setup_container_cwd’:
singularity.c:130: warning: initialization makes pointer from integer without a cast
singularity.c: In function ‘setup_container’:
singularity.c:148: warning: incompatible implicit declaration of built-in function ‘printf’
singularity.c:172: error: redeclaration of ‘image’ with no linkage
singularity.c:145: note: previous declaration of ‘image’ was here
singularity.c:173: warning: assignment makes pointer from integer without a cast
make[3]: *** [singularity.lo] Error 1
make[3]: Leaving directory `singularity-2.3/src/slurm'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `singularity-2.3/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `singularity-2.3/src'
make: *** [all-recursive] Error 1


However, just patching those includes out is not sufficient as it doesn't address re-declaration of image.


singularity.c: In function ‘setup_container_environment’:
singularity.c:65: warning: incompatible implicit declaration of built-in function ‘snprintf’
singularity.c: In function ‘setup_container_cwd’:
singularity.c:129: warning: initialization makes pointer from integer without a cast
singularity.c: In function ‘setup_container’:
singularity.c:147: warning: incompatible implicit declaration of built-in function ‘printf’
singularity.c:171: error: redeclaration of ‘image’ with no linkage
singularity.c:144: note: previous declaration of ‘image’ was here
singularity.c:172: warning: assignment makes pointer from integer without a cast

**Comment by gmkurtzer** [comment]
I can take a stab at this, but the SLURM plugin does need to be updated and @bbockelm (who "owns" that component), has been focused on other things. If anyone wants to try and get to this before I can, it would be greatly appreciated!

Thank you!

**Comment by hsteininger** [comment]
Hi, are there any news about a patch to fix missing header for Slurm Integration PlugIn.

Thanks.

**Comment by mbrt** [comment]
This issue prevents us from upgrading to 2.3.1, because we need the `--with-slurm` and definitely the security fix. Are there any workarounds?

**Comment by votavap** [comment]
I have a patch for the build, but wanted to test the plugin before I submit a pull request. I will see if I can get to it later today or tomorrow unless this is urgent and somebody wants to give it a try.

**Comment by mbrt** [comment]
I'd be happy to test your patch, if this can help to speedup the process

Il 29 giu 2017 20:07, "votavap" <notifications@github.com> ha scritto:

**Comment by votavap** [comment]
@mbrt That would be great!!
https://github.com/votavap/singularity/
It's a fork from the main repo and there is a slurm-plugin-patch-devel branch from the development branch and (if needed) slurm-plugin-patch branch from the master. I managed to build both on CentOS7.3 using gcc 4.8.5.

**Comment by mbrt** [comment]
@votavap the build worked fine for RHEL 7 and the basic singularity functionality is there. I'll now test the slurm functionality. I think the patch is good

**Comment by votavap** [comment]
@mbrt Thank you for testing this!

**Comment by votavap** [comment]
@gmkurtzer I believe this issue can now be closed.

**Comment by votavap** [comment]
@gmkurtzer Or perhaps not till it gets merged to the mater at some point in the future, jumped the gun here a bit ;-) Same with #811

**Comment by gmkurtzer** [comment]
OK, that sounds good. I'll leave this open until it gets merged into master. Thanks!

**Comment by kav2k** [comment]
This can now be closed; it landed in 2.4

**Comment by vsoch** [comment]
Great, thanks @kav2k ,  closing issue.
